### ISSUE TITLE
Upgrade to Pytest 4.2.0

### ISSUE THREAD

**Issue by diox** [mozilla/addons-server#10569]
To upgrade Pytest to 4.2.0 we need to fix a few tests. See the failures in the pyup PR: https://github.com/mozilla/addons-server/pull/10556

**Comment by diox** [comment]
Blocked by https://github.com/pytest-dev/pytest/issues/4704 ; going to pin to pytest<4.2.0 in the meantime

**Comment by blueyed** [comment]
> has been marked as requiring pytest < 4.2.0 for now.

Unfortunately it is `!= 4.2.0`, since I expected this to be an issue in pytest itself.

**Comment by blueyed** [comment]
Would be great if you could try to fix it in pytest-django, given the existing failing test there, and the one you came up with.

**Comment by diox** [comment]
Yeah, I'd like to give it a shot later today but I can't promise anything since I don't know the first thing about pytest-django (or even pytest...).

**Comment by diox** [comment]
@blueyed I gave it a shot for an hour but didn't manage to fix it - I don't know enough about the internals to figure what's wrong without doing a more thorough investigation (what methods were called in which order before compared to now and why). I did notice that just reverting the changes in `pytest/unittest.py` in pytest was enough to fix the issue, but that's not super helpful.

**Comment by blueyed** [comment]
@diox 
Thanks for trying!
Can you check/verify if https://github.com/pytest-dev/pytest-django/pull/700 fixes it for you, please?

**Comment by diox** [comment]
Looks like it does fix it.
